### PR TITLE
Refactor how the images work in the makefile and related scripts

### PR DESCRIPTION
**Pull request by 0sewa0**
# Description
Simplifies a few things related to the makefile:
**I. The confusing way how we set the image/tag that is used during the makefile scripts**

No we have only 3(+1) things that can be set, or set is automatically:
  **+1/ `BRANCH`: The branch of the repo. (your current branch by default)**
         - pipelines don't have the actual git branches/history so we have to set this there manually
    **1/ `IMAGE`: The url to the image, without the tag (`quay...` by default)**
         - we tend to overwrite this in the release-pipeline
    **2/ `TAG`: The tag for the image  (`snapshot` or `snapshot-<sanitised version of $BRANCH>` by default )**
         - we tend to overwrite this in the release-pipeline
    **3/ `IMAGE_URI`: The combination of `$IMAGE`:`$TAG`**
         - just for easier use

**II. Removes unused kuttl scripts**
`kuttl` has not been used for a while

**III. Removes unused operator e2e installation code**
The code is related to installing the operator using the manifests, this is no longer used.

**IV. Removes generated `config/deploy` folder from git**
We still generate the stuff under `config/deploy` but we no longer track it in git.

**MISC**
Used dependencies in the makefiles in a way that we don't do the same thing multiple times, or do things that are not necessary. (sometimes we generated the manifests multiple times, or generated all the manifests when we only needed to generate the CRD) 

## How can this be tested?
Run all the `make` commands that you use, if anything breaks let me know. 😉 

## Checklist
- ~[ ] Unit tests have been updated/added~
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: hack/build/build_image.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+if [[ ! "${1}" ]]; then
+  echo "1 param is not set, should be the image without the tag"
```
Comment (waodim) [inline hack/build/build_image.sh:4]: nitpick: maybe rephrase to `"first param..." `instead of `"1 param..."`?

File: hack/build/build_image.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  exit 5
+fi
+if [[ ! "${2}" ]]; then
+  echo "2 param is not set, should be the tag of the image"
```
Comment (waodim) [inline hack/build/build_image.sh:8]: same as above

File: hack/build/push_image.sh [hunk 3]
```diff
@@ -1,23 +1,20 @@
 #!/bin/bash
 
-if [[ ! "${TAG}" ]]; then
-  echo "TAG variable not set"
-  echo "usage: 'make deploy-local TAG=\"<your-image-tag>\"' or 'make deploy-local-easy'"
+if [[ ! "${1}" ]]; then
+  echo "1 param is not set, should be the image without the tag"
```
Comment (waodim) [inline hack/build/push_image.sh:4]: same as above

File: hack/build/push_image.sh [hunk 4]
```diff
@@ -1,23 +1,20 @@
...
   exit 5
 fi
+if [[ ! "${2}" ]]; then
+  echo "2 param is not set, should be the tag of the image"
```
Comment (waodim) [inline hack/build/push_image.sh:8]: same as above

File: hack/build/push_image.sh [hunk 5]
```diff
@@ -1,23 +1,20 @@
...
---label "quay.expires-after=14d" \
-
-rm -rf third_party_licenses
-
-docker tag "${base_image}" "${out_image}"
+out_image="${IMAGE}${TAG}"
+is_image_present=$(docker images | grep "${out_image}")
+if [ "${is_image_present}" != "" ]; then
+   echo "image is not present, please build it first"
+   exit 5
```
Comment (chrismuellner) [inline hack/build/push_image.sh:18]: why `5`?

File: hack/build/push_image.sh [hunk 6]
```diff
@@ -1,23 +1,20 @@
...
-mkdir -p third_party_licenses
-docker build . -f ./Dockerfile -t "${base_image}" \
---build-arg "GO_LINKER_ARGS=${go_linker_args}" \
---label "quay.expires-after=14d" \
-
-rm -rf third_party_licenses
-
-docker tag "${base_image}" "${out_image}"
+out_image="${IMAGE}${TAG}"
+is_image_present=$(docker images | grep "${out_image}")
```
Comment (chrismuellner) [inline hack/build/push_image.sh:15]: we could use`docker images --type=image <image>` and check for non-zero exit code instead of doing a string comparison.

File: hack/make/images.mk [hunk 7]
```diff
@@ -1,33 +1,25 @@
-MAIN_IMAGE ?= quay.io/dynatrace/dynatrace-operator:snapshot
+IMAGE ?= quay.io/dynatrace/dynatrace-operator
+#Needed for the e2e pipeline to work
+BRANCH ?= $(shell git branch --show-current)
+SNAPSHOT_SUFFIX ?= $(shell echo "${BRANCH}" | sed "s/[^a-zA-Z0-9_-]/-/g")
+ifneq ($(BRANCH), main)
+	TAG ?= snapshot-${SNAPSHOT_SUFFIX}
+else
+	TAG ?= snapshot
+endif
```
Comment (chrismuellner) [inline hack/make/images.mk:9]: this is a lot nicer to read than before

File: hack/make/manifests/config.mk [hunk 8]
```diff
@@ -27,7 +27,7 @@ ifneq ($(shell git branch --show-current | grep "^release-"),)
 	endif
 else ifeq ($(shell git branch --show-current), main)
 	# if the current branch is the main branch
-	CHART_VERSION=snapshot
+	CHART_VERSION=0.0.0-snapshot
```
Comment (chrismuellner) [inline hack/make/manifests/config.mk:30]:
```suggestion
	CHART_VERSION=snapshot
```

File: hack/make/images.mk [hunk 9]
```diff
@@ -1,33 +1,25 @@
...
-endif
+## Pushes an ALREADY BUILT Operator image with a given IMAGE and TAG
+images/build:
+	./hack/build/build_image.sh "${IMAGE}" "${TAG}"
 
-## Builds and pushes an Operator image with a given TAG
+## Builds an Operator image with a given IMAGE and TAG
 images/push:
-	./hack/build/push_image.sh
+	./hack/build/push_image.sh "${IMAGE}" "${TAG}"
```
Comment (chrismuellner) [inline hack/make/images.mk:19]: why do we need to pass environment variables?

File: hack/make/bundle.mk [hunk 10]
```diff
@@ -4,6 +4,8 @@ VERSION ?= 0.0.1
 PLATFORM ?= openshift
 # Needed variable for manifest generation
 OLM ?= false
+# Default bundle image tag
```
Comment (chrismuellner) [inline hack/make/bundle.mk:7]:
```suggestion
# Default bundle image
```

**Review by chrismuellner — CHANGES_REQUESTED** [review]

File: hack/build/push_image.sh (hunk 6 above)
Comment (aorcholski) [reply hack/build/push_image.sh:15]: --type ?
```
[] docker images --type <image>
unknown flag: --type
```

File: hack/make/images.mk (hunk 9 above)
Comment (aorcholski) [reply hack/make/images.mk:19]: IMAGE and TAG are local variables (not seen out of the make script process).

File: hack/build/push_image.sh (hunk 6 above)
Comment (chrismuellner) [reply hack/build/push_image.sh:15]: it's `docker inspect --type=image <image>`

File: hack/make/manifests/config.mk (hunk 8 above)
Comment (0sewa0) [reply hack/make/manifests/config.mk:30]: this is needed for helm, otherwise helm doesn't recognise it as a version

File: hack/make/images.mk (hunk 9 above)
Comment (0sewa0) [reply hack/make/images.mk:19]: its also cleaner IMO (it shows what env is used by what script)

File: hack/build/push_image.sh (hunk 5 above)
Comment (0sewa0) [reply hack/build/push_image.sh:18]: it was like that , I can change it 😅
